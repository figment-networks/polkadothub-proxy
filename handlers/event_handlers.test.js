const eventHandlers = require('./event_handlers');
const {createApi} = require('../tests/shared');

// These actually DO hit the API
describe('integration', () => {
  let call = {request: {height: 2197577}};

  describe('getByHeight', () => {
    it('returns a result that matches snapshot', async () => {
      expect(await eventHandlers.getByHeight(await createApi(), call)).toMatchSnapshot();
    });
  });
});
