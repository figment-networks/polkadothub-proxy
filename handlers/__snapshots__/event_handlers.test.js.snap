// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration getByHeight returns a result that matches snapshot 1`] = `
Object {
  "events": Array [
    Object {
      "data": Array [
        Object {
          "name": "Vec<IdentificationTuple>",
          "value": "[[\\"EDcqtP9vMNrk7PHEbSw1TVgccE3s2mG1xPnbvsQrJ8CWvD2\\",{\\"total\\":\\"0x0000000000000000003cb607bbb46566\\",\\"own\\":428192307900,\\"others\\":[{\\"who\\":\\"DSELWxqt4g3THNfg7TAHhU69eQJo8N2kBbAMuaF636eZCX2\\",\\"value\\":\\"0x0000000000000000003cb5a409744aaa\\"}]}], [\\"EMa2d6yYb3CPxyArDCzsMfngnDKUbRL6QdogfYAPKZmmN1E\\",{\\"total\\":\\"0x0000000000000000003cbc2fa0148b9b\\",\\"own\\":150000000000,\\"others\\":[{\\"who\\":\\"GqPDeJBexSsvEq6gBzrnMCn7KNYbzn9bX9ePCjEDN3EikoV\\",\\"value\\":494995783246335},{\\"who\\":\\"JEcacANpAxTKvrwYCgYRCUVA92sQBvWyCSLY85cQb9TNMJd\\",\\"value\\":3048594856434411},{\\"who\\":\\"DczLnbiH1PAFTegXu1GCp6Py9gujWDoYUPiXVgJYEApp4iY\\",\\"value\\":804220776711627},{\\"who\\":\\"HhM97zfBvM1H7Mee3TDatCsGRNzdoaEodJdjty4bh32NcZb\\",\\"value\\":\\"0x0000000000000000002d49bcf85bece6\\"}]}]]",
        },
      ],
      "description": "[ At the end of the session, at least once validator was found to be offline.]",
      "index": 0,
      "method": "SomeOffline",
      "phase": "initialization",
      "section": "imOnline",
    },
    Object {
      "data": Array [
        Object {
          "name": "Kind",
          "value": "0x696d2d6f6e6c696e653a6f66666c696e",
        },
        Object {
          "name": "OpaqueTimeSlot",
          "value": "0x060f0000",
        },
        Object {
          "name": "bool",
          "value": "true",
        },
      ],
      "description": "[ There is an offence reported of the given \`kind\` happened at the \`session_index\` and,  (kind-specific) time slot. This event is not deposited for duplicate slashes. last,  element indicates of the offence was applied (true) or queued (false).]",
      "index": 1,
      "method": "Offence",
      "phase": "initialization",
      "section": "offences",
    },
    Object {
      "data": Array [
        Object {
          "name": "SessionIndex",
          "value": "3847",
        },
      ],
      "description": "[ New session has happened. Note that the argument is the session index, not the block,  number as the type might suggest.]",
      "index": 2,
      "method": "NewSession",
      "phase": "initialization",
      "section": "session",
    },
    Object {
      "data": Array [
        Object {
          "name": "DispatchInfo",
          "value": "{\\"weight\\":10000000,\\"class\\":\\"Mandatory\\",\\"paysFee\\":\\"No\\"}",
        },
      ],
      "description": "[ An extrinsic completed successfully.]",
      "extrinsicIndex": 0,
      "index": 3,
      "method": "ExtrinsicSuccess",
      "phase": "applyExtrinsic",
      "section": "system",
    },
    Object {
      "data": Array [
        Object {
          "name": "DispatchInfo",
          "value": "{\\"weight\\":10000000,\\"class\\":\\"Mandatory\\",\\"paysFee\\":\\"No\\"}",
        },
      ],
      "description": "[ An extrinsic completed successfully.]",
      "extrinsicIndex": 1,
      "index": 4,
      "method": "ExtrinsicSuccess",
      "phase": "applyExtrinsic",
      "section": "system",
    },
    Object {
      "data": Array [
        Object {
          "name": "DispatchInfo",
          "value": "{\\"weight\\":1000000000,\\"class\\":\\"Mandatory\\",\\"paysFee\\":\\"No\\"}",
        },
      ],
      "description": "[ An extrinsic completed successfully.]",
      "extrinsicIndex": 2,
      "index": 5,
      "method": "ExtrinsicSuccess",
      "phase": "applyExtrinsic",
      "section": "system",
    },
  ],
}
`;
